### PR TITLE
Enrich ConsumerAPI logs with DeviceId and IdentityAddress

### PR DESCRIPTION
**Pull request by Dannyps**
# Readiness checklist

-   [ ] I added/updated unit tests.
-   [ ] I added/updated integration tests.
-   [x] I ensured that the PR title is good enough for the changelog.
-   [x] I labeled the PR.

### CODE REVIEW

File: ConsumerApi/Mvc/Middleware/UserDataLoggingMiddleware.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    public async Task Invoke(HttpContext context)
+    {
+        try
+        {
+            var deviceId = _userContext.GetDeviceId();
+            var identityAddress = _userContext.GetAddress();
+
+            ILogEventEnricher[] enrichers =
```
Comment (tnotheis) [inline ConsumerApi/Mvc/Middleware/UserDataLoggingMiddleware.cs:26]: Isn't it possible to use `var` here?

File: ConsumerApi/Mvc/Middleware/UserDataLoggingMiddleware.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+            using (LogContext.Push(enrichers))
+            {
+                await _next.Invoke(context);
+            }
+        }
+        catch (Exception)
+        {
+            await _next.Invoke(context);
+        }
```
Comment (tnotheis) [inline ConsumerApi/Mvc/Middleware/UserDataLoggingMiddleware.cs:40]: Wouldn't it make sense to use `finally` here?

```suggestion
        try
        {
            var deviceId = _userContext.GetDeviceId();
            var identityAddress = _userContext.GetAddress();

            ILogEventEnricher[] enrichers =
            {
                new PropertyEnricher("deviceId", deviceId),
                new PropertyEnricher("identityAddress", identityAddress)
            };
            
            LogContext.Push(enrichers)
        }
        finally
        {
            await _next.Invoke(context);
        }
```

File: ConsumerApi/appsettings.override.json [hunk 3]
```diff
@@ -135,9 +135,9 @@
         "WriteTo": [
             {
                 "Name": "Console",
-                "Args": {
-                    "outputTemplate": "{Timestamp:yyyy-MM-dd HH:mm:ss.fff}|{Level} - CorrelationId:{CorrelationId} - RequestId:{RequestId} - RequestPath:{RequestPath}{NewLine}    {SourceContext}{NewLine}    {Message}{NewLine}{Exception}"
-                }
+              "Args": {
+                "outputTemplate": "{Timestamp:yyyy-MM-dd HH:mm:ss.fff}|{Level} - CorrelationId:{CorrelationId} - RequestId:{RequestId} - RequestPath:{RequestPath} - DeviceId: {deviceId} - Address: {identityAddress} {NewLine}    {SourceContext}{NewLine}    {Message}{NewLine}{Exception}"
```
Comment (tnotheis) [inline ConsumerApi/appsettings.override.json:139]:
```suggestion
                "outputTemplate": "{Timestamp:yyyy-MM-dd HH:mm:ss.fff}|{Level} - CorrelationId:{CorrelationId} - RequestId:{RequestId} - RequestPath:{RequestPath} - DeviceId:{deviceId} - Address:{identityAddress} {NewLine}    {SourceContext}{NewLine}    {Message}{NewLine}{Exception}"
```

**Review by tnotheis — CHANGES_REQUESTED** [review]

File: ConsumerApi/Mvc/Middleware/UserDataLoggingMiddleware.cs (hunk 1 above)
Comment (Dannyps) [reply ConsumerApi/Mvc/Middleware/UserDataLoggingMiddleware.cs:25]: It is if we do the following:

```c#
var enrichers = new ILogEventEnricher[]
{
    new PropertyEnricher("deviceId", deviceId),
    new PropertyEnricher("identityAddress", identityAddress)
};
```

Is that preferred?

File: ConsumerApi/Mvc/Middleware/UserDataLoggingMiddleware.cs [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            var deviceId = _userContext.GetDeviceId();
+            var identityAddress = _userContext.GetAddress();
+            ILogEventEnricher[] enrichers =
+            {
+                new PropertyEnricher("deviceId", deviceId),
+                new PropertyEnricher("identityAddress", identityAddress)
+            };
+
+            using var disposable = LogContext.Push(enrichers);
```
Comment (tnotheis) [inline ConsumerApi/Mvc/Middleware/UserDataLoggingMiddleware.cs:31]:
```suggestion
            using var _ = LogContext.Push(enrichers);
```

**Review by tnotheis — CHANGES_REQUESTED** [review]

**Review by tnotheis — DISMISSED** [review]

**Review by tnotheis — APPROVED** [review]

**Pull request merged**